app/src/features/FhirResourceTree/resourceTreeSlice.ts

```diff
@@ -52,11 +63,67 @@ const resourceTreeSlice = createSlice({
         if (node) node.children = data.children;
       }
     },
+    setAttributeNodes: (
+      state,
+      { payload }: PayloadAction<{ attributeNodes: ElementNode[] }>
+    ) => {
+      const { attributeNodes } = payload;
+      const { attributeNodesInTree, root } = state;
+      if (root) {
+        const nodesToRemove = differenceBy(
+          attributeNodesInTree,
+          attributeNodes,
+          (node) => node.path
+        );
+        const nodesToAdd = differenceBy(
+          attributeNodes,
+          attributeNodesInTree,
+          (node) => node.path
+        );
+
+        // Add attribute nodes to the tree
+        nodesToAdd.forEach((node) => {
+          const parent = getNode(
+            "path",
+            computePathWithoutIndexes(node),
+            root as ElementNode
+          );
+          if (
+            parent &&
+            !parent.children.some(({ path }) => path === node.path)
+          ) {
+            parent.children.push(node);
+            attributeNodesInTree.push(node);
```

Comment:
why do we need `attributeNodesInTree` at all ? To differentiate nodes generated from definitions from those generated from the user inputs ? Can we have this information in the nodes themselves?

Reply:
We need to handle nodes generated from attributes in a different thread than the tree builder itself, to prevent a whole tree reload.
Then, in order to regroup attribute add/delete actions into a single one, I decided to cache the whole attribute nodes array in the store to compute differential when the entry has changed.
I think, if we only want to add attributes, that this would work. But when attributes are deleted, it would be a mess to find back those nodes.. Right?

Did I answer your question?

Reply:
yep thanks!